### PR TITLE
MKL: Removing visited_node hash table - fixing multinode shape mismatch issue

### PR DESCRIPTION
**Pull request by nhasabni**
This PR fixes an issue that arises in multinode setup. MklLayoutRewritePass
maintains a hash table for visited nodes, and the hash table is part of the
pass and used for every graph being rewritten. But it looks like in multinode
setup, multiple graphs may be processed simulteneously leading to incorrect
modifications to the hash table. So removing the hash table as we do not
really need it.

### CODE REVIEW

**Comment by Mazecreator** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  mazecreator@gmail.com
    Domain mazecreator.com has exceeded the max emails per hour (29/25 (115%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o11.sgmail.github.com ([167.89.101.202]:2958)
	by server2.lowesthostingrates.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.89)
	(envelope-from <bounces+848413-e6d9-mazecreator=mazecreator.com@sgmail.github.com>)
	id 1ddgsG-0003QU-I7
	for mazecreator@mazecreator.com; Fri, 04 Aug 2017 12:59:30 -0500
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=NvO5B3vmfojaNDd6C0u+AIQthRc=; b=aXJn+ZuGpg1OuPNN
	Mx89X13fUEwaltVuCh+YjBGW3tn3J7/K24ZAOrjAZh95IHOpzccZPj+8xlaRmM85
	I2QWbhMR8dZev3cVcBH0swyxnGgNf15lVYB/Prl4N+VZmjJcznXjirFqrTb9VZ++
	3cx+MSW97rkJRUx0eTt5nfObVXk=
Received: by filter0553p1mdw1.sendgrid.net with SMTP id filter0553p1mdw1-3629-5984B867-4F
        2017-08-04 18:09:43.709876163 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0026p1mdw1.sendgrid.net (SG) with ESMTP id h2CLpfkzRRmoRQ6eG1yeaA
	for <mazecreator@mazecreator.com>; Fri, 04 Aug 2017 18:09:43.653 +0000 (UTC)
Date: Fri, 04 Aug 2017 18:09:43 +0000 (UTC)

**Comment by Mazecreator** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  mazecreator@gmail.com
    Domain mazecreator.com has exceeded the max emails per hour (30/25 (120%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o10.sgmail.github.com ([167.89.101.201]:60883)
	by server2.lowesthostingrates.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.89)
	(envelope-from <bounces+848413-e6d9-mazecreator=mazecreator.com@sgmail.github.com>)
	id 1ddgsK-0003Qg-03
	for mazecreator@mazecreator.com; Fri, 04 Aug 2017 12:59:34 -0500
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=JojCJuC50pBQjtqpufyp3oTzkgo=; b=G1qaRLoz7mT9Zfy0
	bGkvcKKbz9xVz88fGPBvIdp9s5GcEEx1f08Jw11NdxZNFSQvqMoUREWQSmmhqmCs
	2F5kV4ogg2RkgskNNPYZkqA/ilIjennM1o4VhRR/B/7fSOYL/qGgq5vv8pgyc9lR
	iIy5hpU49wS9PdhhpWYRF3Ay7kc=
Received: by filter0648p1mdw1.sendgrid.net with SMTP id filter0648p1mdw1-13987-5984B86C-58
        2017-08-04 18:09:48.49781257 +0000 UTC
Received: from github-smtp2b-ext-cp1-prd.iad.github.net (github-smtp2b-ext-cp1-prd.iad.github.net [192.30.253.17])
	by ismtpd0036p1mdw1.sendgrid.net (SG) with ESMTP id Iy8FxhLcT9y55P-CEsQ8iQ
	for <mazecreator@mazecreator.com>; Fri, 04 Aug 2017 18:09:48.539 +0000 (UTC)
Date: Fri, 04 Aug 2017 18:09:48 +0000 (UTC)

File: tensorflow/core/graph/mkl_layout_pass.cc [hunk 1]
```diff
@@ -1087,15 +1067,13 @@ void MklLayoutRewritePass::GetNodeProducingMklTensor(std::unique_ptr<Graph>* g,
...
-    DataType T;
-    TF_CHECK_OK(GetNodeAttr(n->def(), "T", &T));
-    // If this op has been rewritten, then its name must have been same as
-    // Mkl op.
-    CHECK_EQ(mkl_op_registry::IsMklOp(n->type_string(), T), true);
+
+  // If this is an MKL op, then it will create extra output for MKL layout.
+  DataType T;
+  if (GetNodeAttr(n->def(), "T", &T).ok() &&
+      mkl_op_registry::IsMklOp(n->type_string(), T)) {
```
Comment (rmlarsen) [inline tensorflow/core/graph/mkl_layout_pass.cc]: Is there a potential performance hit here? Was the hash table there to compensate for IsMklOp being slow? If not, this is obviously fine.

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

File: tensorflow/core/graph/mkl_layout_pass.cc (hunk 1 above)
Comment (nhasabni) [inline tensorflow/core/graph/mkl_layout_pass.cc]: No, there is no performance penalty. IsMklOk function is just a simple predicate, so it is not at all slow.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Pull request merged**